### ISSUE TITLE
Resize while preserving aspect ratio to specified dimensions, using fill if necessary

### ISSUE THREAD

**Issue by slayful** [kohler/gifsicle#138]
I've tried gifsicle on a sample of images and it is working amazing. It's fast, transforms images that other libs can't handle and the results are compatible with webP conversion.

There's however one feature I'm missing, which is resizing to specified dimensions (both width and height) while preserving aspect ratio.
As far as I can tell, existing modes produce images that either preserve aspect ratio and don't *exceed* the dimensions, satisfying at least one (width or height) of them OR break aspect ratio and fit the dimensions exactly.

Basically what I'm looking for is gifsicle to fill the missing pixels to match the dimensions. Here's an example with a blue fill: https://ibb.co/z5GRnYb

Is this something that could be easily added?

**Comment by marcing** [comment]
I'm looking forward to the same feature. It's called resize with padding I believe.

Thank you.

**Comment by slayful** [comment]
https://github.com/kohler/gifsicle/pull/144 this should resolve this :)
